library/goptions.mli

```diff
@@ -149,6 +149,8 @@ val declare_string_option_and_ref : (value:string -> (unit -> string)) opt_decl
 val declare_stringopt_option_and_ref : (unit -> string option) opt_decl
 val declare_interpreted_string_option_and_ref :
   (value:'a -> (string -> 'a) -> ('a -> string) -> (unit -> 'a)) opt_decl
+val declare_synterp_string_option_and_ref :
+  (value:'a -> (string -> 'a) -> ('a -> string) -> (unit -> 'a)) opt_decl
```

Comment:
The "interpreted" here refers to the string being interpreted into a `'a`, not to the stage IMO
so the API should be modified to have the optional `stage` argument, not adding a new function.

Reply:
Not trivial without breaking compatibility: the `opt_decl` abstraction exposes only named arguments. So adding an optional probably also requires adding a `unit` argument. Or do you think we can make this precise function no longer use the `opt_decl` signature?

Reply:
Adding a unit argument sounds fine